src/Altinn.Apps/AppFrontend/react/altinn-app-frontend/src/shared/containers/Presentation.tsx

```diff
@@ -51,9 +51,16 @@ const PresentationComponent = (props: IPresentationProvidedProps) => {
 
   const handleModalCloseButton = () => {
     const origin = window.location.origin;
-    if (window) {
-      window.location.href = returnUrlToMessagebox(origin, party.partyId);
+    const returnUrl = returnUrlFromQueryParameter();
+    let windowHref = '';
+
+    if (returnUrl) {
+      windowHref = returnUrl;
+    } else {
+      windowHref = returnUrlToMessagebox(origin, party.partyId);
     }
+
+    window.location.href = windowHref;
```

Comment:
Could simply logic to
```
 window.location.href = returnUrlFromQueryParameter() || returnUrlToMessagebox(origin, party.partyId);
```

Also a test for this would be neat.

Reply:
I'll have a look at that

Reply:
Fixed and added 2 tests